data-raw/create_sdtms_data.R

```diff
@@ -0,0 +1,134 @@
+# Load the XLSX file with metadata
+library(readxl)
+
+specs <- read_excel("inst/extdata/sdtms-specs.xlsx")
+
+# Helper function to get attribute for a column, with default as "undocumented field"
+get_attr <- function(data, col_name) {
+  att <- attr(data[[col_name]], "label")
+  if (is.null(att) || att == "null") {
+    att <- "undocumented field"
+  }
+  return(att)
+}
+
+# Helper function to check the length of a string and append " # nolint" if it exceeds 90 characters
+check_and_append_nolint <- function(text) {
+  if (nchar(text) > 90) {
+    text <- paste0(text, " # nolint")
+  }
+  return(text)
+}
+
+# Define the `write_doc` function, now with `dataset_description`
+write_doc <-
+  function(data,
+           dataset_name,
+           dataset_label = "No label available",
+           dataset_description = "No description available",
+           dataset_author = NULL,
+           dataset_source = "No source available") {
+    # Check and adjust lengths of metadata fields
+    dataset_label <- check_and_append_nolint(dataset_label)
+    dataset_description <- check_and_append_nolint(dataset_description)
+    dataset_source <- check_and_append_nolint(dataset_source)
+    if (!is.null(dataset_author)) {
+      dataset_author <- check_and_append_nolint(dataset_author)
+    }
+
+    # Create documentation for the current dataset
+    doc_string <- paste(
+      sprintf("# This file is automatically generated."),
```

Comment:
Could we add more information? For example:
```suggestion
      sprintf("# This file is automatically generated by data-raw/create_sdtms_data.R."),
      sprintf("# For updating it please edit inst/extdata/sdtms-specs.json and rerun create_sdtms_data.R."),
```